### PR TITLE
chore(github): skip some heavy jobs when PR is not related

### PR DESCRIPTION
**Pull request by jsvisa**
I've seen in this PR https://github.com/paradigmxyz/reth/pull/6665, I noticed that even though it was just a change to one file in the Grafana dashboard, it still triggered almost all the github action tasks, which is resource-intensive. Additionally, it could potentially block other merge tasks due to failures, so try to skip some github jobs if not related.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -110,21 +112,11 @@ jobs:
           components: rustfmt
       - run: cargo fmt --all --check
 
-  grafana:
```
Comment (onbjerg) [inline .github/workflows/lint.yml:113]: please don't remove this, we use it to ensure formatting

**Review by onbjerg — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (jsvisa) [reply .github/workflows/lint.yml:113]: I move this into an independent job, launch only if the grafana files are modified

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
reasonable

**Pull request merged**